RELEASE: first iteration of web-application released

EyesHead opened the pull request:
Здесь я использовал библиотеку mapstruct для маппинга dtoшек, так же оставил нетронутыми Request и Booking логику, поскольку она не требуется по ТЗ. По этой же причине я оставил в ItemResponseDto айдишник owner, а не объект UserResponseDto (это трудно реализовать в рамках данного спринта без знаний о JPA и ORM)

Моя цель от данного спринта как и всегда - получить максимум. Так что если какая-то практика, используемая мной часто в коде - плохая, то скажите мне, я запомню это навсегда.

Commit:
RELEASE: first iteration of web-application released

Commit:
fixed: pom.xml restored to template version

Commit:
style: checkstyle failed fixed

Commit:
pom.xml updated

bismuthD commented on src/main/java/ru/practicum/shareit/user/mapper/UserDtoMapper.java:
```diff
@@ -0,0 +1,16 @@
+import ru.practicum.shareit.user.dto.UserResponseDto;
+import ru.practicum.shareit.user.model.User;
+
+@Mapper(componentModel = "spring")
```
верно выбрал аннотацию для маппера, так же вместо нее можно использовать @NoArgsConstructor(access = AccessLevel.PRIVATE)

bismuthD commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,85 @@
+    private final ItemService itemService;
+
+    @PostMapping
+    public ItemResponseDto createItemForUser(@RequestHeader(value = "X-Sharer-User-Id") long userId,
```
X-Sharer-User-Id дублируется, следует вынести значение в константу, для этого лучше выделить служебный класс, в который будешь добавлять все константы по проекту

EyesHead replied:
Это интересное решение. Но наверное лучше даже сразу в property закинуть такое значение, чтобы из одного файла всеми пропертями управлять

bismuthD commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,85 @@
+    @PostMapping
+    public ItemResponseDto createItemForUser(@RequestHeader(value = "X-Sharer-User-Id") long userId,
+                                             @Valid @RequestBody ItemPostDto itemCreate) {
+        log.info("Received request to create item for userId: {}", userId);
```
логирование начала выполнения и окончания - это перебор)
Достаточно логировать или начало операции (создается, обновляется, удаляется то-то) или факт выполнения метода (что-то создано, обновлено, удалено)

bismuthD reviewed (commented):
Отличная работа, оставил пару комментариев на доработку

Commit:
refactor:
the request header parameter at ItemController was moved to a separate class - ItemControllerValues;
redundant loggers was removed
